homeassistant/components/airly/__init__.py

```diff
@@ -76,6 +77,15 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             entry, unique_id=f"{latitude}-{longitude}"
         )
 
+    # identifiers in device_info should use Tuple[str, str, str] type, but latitude and
+    # longitude are float, so we convert old device entries to use correct types
+    device_registry = await async_get_registry(hass)
+    old_ids = (DOMAIN, latitude, longitude)
+    device_entry = device_registry.async_get_device({old_ids})
+    if device_entry and entry.entry_id in device_entry.config_entries:
+        new_ids = (DOMAIN, str(latitude), str(longitude))
+        device_registry.async_update_device(device_entry.id, new_identifiers={new_ids})
```

Comment:
The `identifiers` in` device_info` should be `Tuple [str, str, str]`, but now it is `Tuple [str, float, float]`. So I added migrating device entry to the correct type.